engine/src/main/java/org/terasology/input/InputSystem.java

```diff
@@ -420,8 +428,39 @@ private void processControllerInput(float delta) {
         }
     }
 
+
+    /**
+     * Simulated key strokes: To simulate input from a keyboard, we simply have to extract the Input associated to the action
+     * and this function adds it to the keyboard's input queue.
+     * @param key The key to be simulated.
+     */
+    public void simulateSingleKeyStroke(Input key) {
+        /* TODO: Perhaps there is a better way to extract the character.
+            All the simulate functions extract keyChar by getting the first character from it's display string.
+            While it works for normal character buttons, might not work for special buttons if required later.
+        */
+        char keyChar = key.getDisplayName().charAt(0);
```

Comment:
That's what I thought, but what about the case where the user has move forward set to the up arrow on his keyboard? It will return a string "UP" and not a single character. Also, no matter what the value of keyChar is in the KeyboardAction instance is, it works. I added this since there was no constructor without it. So, I kinda left it the way it is right now.